corehq/apps/sso/tasks.py

```diff
@@ -99,3 +101,58 @@ def send_idp_cert_expires_reminder_emails(num_days):
                 f"Failed to send cert reminder email for IdP {idp}: {exc!s}",
                 exc_info=True,
             )
+
+
+@periodic_task(run_every=crontab(minute=0, hour=14), acks_late=True)
+def send_api_token_expiration_reminder():
+    """
+    Sends reminder emails for IDP api secret expiring N days from today.
+    """
+    for num_days in IDP_API_SECRET_EXPIRES_REMINDER_DAYS:
+        send_api_token_expiration_reminder_emails(num_days)
+
+
+def send_api_token_expiration_reminder_emails(num_days):
+    """
+    Sends a reminder email to the enterprise admin email addresses specified on
+    the IdP's owner account for any IdPs with api secret that will expire on
+    the date `num_days` from today, and have enable_user_deactivation is set to True.
+
+    :param num_days: Query IdP api secret expiring `num_days` from today.
+    """
+    today = datetime.datetime.utcnow().date()
+    date_in_n_days = today + datetime.timedelta(days=num_days)
+    day_after_that = date_in_n_days + datetime.timedelta(days=1)
+    queryset = IdentityProvider.objects.filter(
+        is_active=True,
+        enable_user_deactivation=True,
+        date_api_secret_expiration__gte=date_in_n_days,
+        date_api_secret_expiration__lt=day_after_that,
+    )
+    for idp in queryset.all():
+        context = get_api_secret_expiration_email_context(idp)
+        if not context["to"]:
+            log.error(f"no admin email addresses for IdP: {idp}")
+        try:
+            for send_to in context["to"]:
+                send_html_email_async.delay(
+                    context["subject"],
+                    send_to,
+                    context["html"],
+                    text_content=context["plaintext"],
+                    email_from=context["from"],
+                    bcc=context["bcc"],
+                )
+                log.info(
+                    "Sent %(num_days)s-day api secret expiration reminder "
+                    "email for %(idp_name)s to %(send_to)s." % {
+                        "num_days": num_days,
+                        "idp_name": idp.name,
+                        "send_to": send_to,
+                    }
+                )
+        except Exception as exc:
+            log.error(
```

Comment:
i feel like this also should be a `notify_exception`. we want to fail loudly around not being able to send these messages

Reply:
Sorry, made a mistake and fixed here: dab3e08e78ff875be2dfc501d878e32b93672bbf